### PR TITLE
create an ID type in pods package for pod IDs

### PR DESCRIPTION
**Pull request by mpuncel**
This will increase type safety when passing around pod IDs from package
to package. For instance, the kp package can provide some safety around
constructing keys by differentiating the pod id type from the hostname
type.

### CODE REVIEW

**Comment by mpuncel** [comment]
I intend to follow this up with a PR that makes various `pkg/kp` operations use this type, e.g. `IntentPath` and `SetPod`. The PR was huge though so I decided to break it up

File: bin/p2-bootstrap/bootstrap.go [hunk 1]
```diff
@@ -39,7 +40,7 @@ func main() {
 		if err != nil {
 			log.Fatalf("Could not get consul manifest: %s", err)
 		}
-		consulPod = pods.NewPod(consulManifest.ID(), pods.PodPath(*podRoot, consulManifest.ID()))
+		consulPod = pods.NewPod(string(consulManifest.ID()), pods.PodPath(*podRoot, consulManifest.ID()))
```
Comment (anthonybishopric) [inline bin/p2-bootstrap/bootstrap.go]: Make `NewPod` take a `types.PodID`?

File: bin/p2-bootstrap/bootstrap.go [hunk 2]
```diff
@@ -128,7 +129,7 @@ func VerifyConsulUp(timeout string) error {
 	}
 }
 
-func VerifyReality(waitTime time.Duration, consulID, agentID string) error {
+func VerifyReality(waitTime time.Duration, consulID types.PodID, agentID types.PodID) error {
```
Comment (petertseng) [inline bin/p2-bootstrap/bootstrap.go]: can say `types.PodID` just once, unless this was deliberate?

Comment (mpuncel) [inline bin/p2-bootstrap/bootstrap.go]: I personally don't really like reusing the type, happy to put it back if people disagree

File: bin/p2-bootstrap/bootstrap.go (hunk 1 above)
Comment (mpuncel) [inline bin/p2-bootstrap/bootstrap.go]: The call sites for NewPod are pretty evenly split between callers wanting to pass a string vs an id pulled from some other manifest. I felt like it's philosophically better to make NewPod do the cast for you, but if you feel strongly I can change it

Comment (anthonybishopric) [inline bin/p2-bootstrap/bootstrap.go]: That makes sense, however I'd argue that it's weird to cast back to `string` if you do have a manifest object though. I'd still be in favor of making `NewPod` take a `types.PodID`

Comment (mpuncel) [inline bin/p2-bootstrap/bootstrap.go]: okay i'll make the change

File: bin/p2-bootstrap/bootstrap.go (hunk 2 above)
Comment (petertseng) [inline bin/p2-bootstrap/bootstrap.go]: OK. I don't care strongly either way

**Comment by bwester** [comment]
:ship:

**Comment by anthonybishopric** [comment]
:+1: pending comments

**Pull request merged**